util/file_util.py

```diff
@@ -0,0 +1,922 @@
+      file name combined from path, suffix and extension.
+    '''
+    # Create the file directory if it doesn't exist.
+    file_makedirs(file_path)
+    file_prefix, ext = os.path.splitext(file_path)
+    if file_prefix.endswith(suffix):
+        # Suffix already present in name, ignore it.
+        suffix = ''
+    # Set the file extension
+    if file_ext and file_ext[0] != '.':
+        file_ext = '.' + file_ext
+    return file_prefix + suffix + file_ext
+
+
+def file_makedirs(filename: str) -> str:
+    '''Creates the directory for the filename and returns the directory.
+    Only supports files on the local file system.
+
+    Args:
+      filename: name of the file with the directory.
+    Returns:
+      the directory for the filename, created if not avilable.
+    '''
+    if filename:
+        dirname = os.path.dirname(filename)
+        if dirname and file_is_local(dirname):
+            os.makedirs(dirname, exist_ok=True)
+            return dirname
+    return ''
+
+
+def file_copy(src_filename: str, dst_filename: str = '') -> str:
+    '''Copies over the src_file into the dst_file and returns the filename.
+    Supports both local files, GCS files and spreadsheets.
+
+    Args:
+      src_filename: string filename of file to be copied
+      dst_filename: string filename of file to be copied into
+        If not set, creates a local file with suffix <src_filename>-copy.<ext>
+    Returns:
+      the destination file into which source file content was copied into.
+    '''
+    if not dst_filename:
+        # Create a filename for destination with a '-copy' suffix.
+        basename = os.path.basename(src_filename)
+        dst_filename = file_get_name(basename, suffix='-copy')
+    # Open both files and copy content
+    file_makedirs(dst_filename)
+    with FileIO(filename=src_filename, mode='rb', use_tempfile=False) as src:
+        with FileIO(filename=dst_filename, mode='wb',
+                    use_tempfile=False) as dst:
+            _copy_file_chunks(src, dst)
+    return dst_filename
+
+
+def file_load_csv_dict(filename: str,
+                       key_column: str = None,
+                       value_column: str = None,
+                       delimiter: str = ',',
+                       config: dict = {}) -> dict:
+    '''Returns a dictionary loaded from a CSV file.
+  Each row is added to the dict with value from column 'key_column' as key
+  and  value from 'value_column.
+  Args:
+    filename: csv file name to be loaded into the dict.
+      it can be a comma separated list of file patterns as well.
+    key_column: column in the csv to be used as the key for the dict
+      if not set, uses the first column as the key.
+    value_column: column to be used as value in the dict.
+      If not set, value is a dict of all remaining columns.
+    config: dictionary of aggregation settings in case there are
+      multiple rows with the same key.
+      refer to _dict_aggregate_values() for config settings.
```

Comment:
Might be useful to keep low-level utils separated from higher-level function like aggregation?

Reply:
ok, will change to overwrite value for existing keys instead.

Reply:
Oh the functionality is good, I was just pointing out "file_util" is perhaps not the best place for it?

Reply:
Moved to aggregation_utli